gym_jadx/envs/jadx_env.py

```diff
@@ -0,0 +1,412 @@
+        """
+        progress_vector = []
+        for button in self.__all_buttons:
+            if button.reward_given:
+                progress_vector.append(1)
+            else:
+                progress_vector.append(0)
+        return np.array(progress_vector)
+
+    def step(self, action: ndarray) -> (ndarray, int, bool):
+        reward = 0
+        number_of_windows_to_be_removed = 0
+
+        # Click on windows starting with the topmost window down to the window at the bottom
+        # Continue clicking only while the current window was not clicked AND
+        # the current window is not modal
+        for i in range(-1, -len(self.__windows) - 1, -1):
+            index = i + number_of_windows_to_be_removed
+            reward, window_includes_point, clicked_child_component_matrix, clicked_child_component_coords = \
+                self.__windows[index].click(action)
+            if window_includes_point:
+                if clicked_child_component_matrix is not None:
+                    # Blit the changed component directly on the frame buffer, only if the clicked component is
+                    # in the topmost window
+                    if not self.__should_re_stack:
+                        MatrixUtils.blit_image_inplace(self.__frame_buffer,
+                                                       clicked_child_component_matrix,
+                                                       clicked_child_component_coords[0],
+                                                       clicked_child_component_coords[1])
+                break
+            else:
+                if self.__windows[index].modal:
+                    break
+                elif self.__windows[index].auto_close:
+                    # Save the removed window for future reference and continue with the loop
+                    removed_window = self.__remove_window()
+                    self.__windows_to_be_removed.append(removed_window)
+                    number_of_windows_to_be_removed += 1
+
+        # Redraw all windows by stacking them up from the bottom to the top, if needed
+        if self.__should_re_stack:
+            self.__frame_buffer = self.__stack_windows()
+
+        # Reset internal state
+        self.__should_re_stack = False
+        self.__windows_to_be_removed.clear()
+        return self.__frame_buffer, reward, self.__done, None
+
+    def __add_window(self, window: Window):
+        self.__windows.append(window)
+        MatrixUtils.blit_image_inplace(self.__frame_buffer,
+                                       window.current_matrix,
+                                       window.relative_coordinates[0], window.relative_coordinates[1])
+
+    def __remove_window(self) -> Window:
+        removed = self.__windows.pop()
+        removed.reset()
+        self.__windows[-1].reset()
+        self.__should_re_stack = True
+        return removed
+
+    def __change_visibility(self, drawable: Drawable, value: bool):
```

Comment:
Yes I didn't use this functionality but I left it there in case it can be used in the future. However, I haven't tested
it thoroughly, so it is probably buggy and having an unused feature also reduces clarity, so I completely removed this
feature in the last commit.